Correct rover param download

stephendade opened the pull request:
For the rover param downloads, there was an inconsistency between using "APMrover2.xml" and "Rover.xml", leading to the param module not using the correct xml file.

I've standardised on "APMrover2.xml"

peterbarker commented:
On Tue, 8 Apr 2025, Stephen Dade wrote:

> For the rover param downloads, there was an inconsistency between using
> "APMrover2.xml" and "Rover.xml", leading to the param module not using the
> correct xml file.
> 
> I've standardised on "APMrover2.xml"

Any particular reason?  I believe I added APMrover2.xml as a symlink on 
the server, expecting to kill it eventually!

stephendade commented:
> Any particular reason?  I believe I added APMrover2.xml as a symlink on
the server, expecting to kill it eventually!

I was going off https://autotest.ardupilot.org/Parameters/, where it was the other way around -  Rover was symlinked to APMRover2.

Happy to change to PR to using Rover.xml - makes more sense that way

peterbarker commented:
> > Any particular reason?  I believe I added APMrover2.xml as a symlink on
> > the server, expecting to kill it eventually!
> 
> I was going off https://autotest.ardupilot.org/Parameters/, where it was the other way around - Rover was symlinked to APMRover2.
> 
> Happy to change to PR to using Rover.xml - makes more sense that way

Yep, thanks.

I've also added symlinks so we can refer to just Copter / Plane / Sub:
```
lrwxrwxrwx 1 autotest autotest    5 May 30  2020 APMrover2 -> Rover
drwxrwxr-x 2 autotest autotest 4096 Apr  9 02:36 AP_Periph
drwxrwxr-x 2 autotest autotest 4096 Apr  9 02:36 AntennaTracker
drwxrwxr-x 2 autotest autotest 4096 Apr  9 02:35 ArduCopter
drwxrwxr-x 2 autotest autotest 4096 Apr  9 02:35 ArduPlane
drwxrwxr-x 2 autotest autotest 4096 Apr  9 02:35 ArduSub
drwxrwxr-x 2 autotest autotest 4096 Apr  9 02:36 Blimp
lrwxrwxrwx 1 autotest autotest   10 Apr  9 23:50 Copter -> ArduCopter
drwxr-xr-x 2 autotest autotest 4096 May 17  2021 Heli
lrwxrwxrwx 1 autotest autotest    9 Apr  9 23:49 Plane -> ArduPlane
-rw-rw-r-- 1 autotest autotest  108 May 17  2021 README
drwxrwxr-x 2 autotest autotest 4096 Apr  9 02:35 Rover
drwxrwxr-x 2 autotest autotest 4096 Dec 14  2022 SITL
lrwxrwxrwx 1 autotest autotest    7 Apr  9 23:50 Sub -> ArduSub
drwxrwxr-x 7 autotest autotest 4096 Sep  4  2024 versioned
```

stephendade commented:
Fixed up to the new download locations.

I also removed some old ``param diff`` code that attempted to grab the ``defaults.parm`` from ``.mavproxy``. Not sure when that was a thing?

peterbarker commented on .github/workflows/windows_build.yml:
```diff
@@ -45,10 +45,10 @@ jobs:
+          $wc.DownloadFile('https://autotest.ardupilot.org/Parameters/Rover/apm.pdef.xml', 'Parameters\Rover.xml')
+          $wc.DownloadFile('https://autotest.ardupilot.org/Parameters/Copter/apm.pdef.xml', 'Parameters\Copter.xml')
+          $wc.DownloadFile('https://autotest.ardupilot.org/Parameters/Plane/apm.pdef.xml', 'Parameters\Plane.xml')
+          $wc.DownloadFile('https://autotest.ardupilot.org/Parameters/Sub/apm.pdef.xml', 'Parameters\Sub.xml')
```
Not your bug, but missing Blimp here

peterbarker commented on windows/MAVProxyWinBuild.bat:
```diff
@@ -53,10 +53,10 @@ rem -----Create version Info-----
+powershell.exe "Start-BitsTransfer -Source 'http://autotest.ardupilot.org/Parameters/Rover/apm.pdef.xml' -Destination 'Parameters\Rover.xml'"
+powershell.exe "Start-BitsTransfer -Source 'http://autotest.ardupilot.org/Parameters/Copter/apm.pdef.xml' -Destination 'Parameters\Copter.xml'"
+powershell.exe "Start-BitsTransfer -Source 'http://autotest.ardupilot.org/Parameters/Plane/apm.pdef.xml' -Destination 'Parameters\Plane.xml'"
+powershell.exe "Start-BitsTransfer -Source 'http://autotest.ardupilot.org/Parameters/Sub/apm.pdef.xml' -Destination 'Parameters\Sub.xml'"
```
Missing Blimp here, too

stephendade commented:
Great. I've added the Blimp param downloads now as well :)
